application/src/main/java/org/togetherjava/tjbot/commands/tags/TagManageCommand.java

```diff
@@ -285,6 +344,62 @@ private boolean isWrongTagStatusAndHandle(@NotNull TagStatus requiredTagStatus,
         return false;
     }
 
+    private void logAction(@NotNull Subcommand subcommand, @NotNull Guild guild,
+            @NotNull User author, @NotNull TemporalAccessor timestamp, @NotNull String id,
+            @Nullable String newContent, @Nullable String previousContent) {
+
+        if (EnumSet
+            .of(Subcommand.CREATE, Subcommand.CREATE_WITH_MESSAGE, Subcommand.EDIT,
+                    Subcommand.EDIT_WITH_MESSAGE)
+            .contains(subcommand) && newContent == null) {
+            throw new IllegalArgumentException(
+                    "newContent is null even though the subcommand should supply a value.");
+        }
+
+        if (EnumSet.of(Subcommand.EDIT, Subcommand.EDIT_WITH_MESSAGE, Subcommand.DELETE)
+            .contains(subcommand) && previousContent == null) {
+            throw new IllegalArgumentException(
+                    "previousContent is null even though the subcommand should supply a value.");
+        }
+
+        switch (subcommand) {
+            case CREATE -> ModAuditLogWriter.write("Tag-Manage Create",
+                    String.format(LOG_EMBED_DESCRIPTION, subcommand.getActionVerb(), id), author,
+                    timestamp, guild, new ModAuditLogWriter.Attachment(CONTENT_FILE_NAME,
+                            Objects.requireNonNull(newContent)));
+
+            case CREATE_WITH_MESSAGE -> ModAuditLogWriter.write("Tag-Manage Create with message",
+                    String.format(LOG_EMBED_DESCRIPTION, subcommand.getActionVerb(), id), author,
+                    timestamp, guild, new ModAuditLogWriter.Attachment(CONTENT_FILE_NAME,
+                            Objects.requireNonNull(newContent)));
+
+            case EDIT -> ModAuditLogWriter.write("Tag-Manage Edit",
+                    String.format(LOG_EMBED_DESCRIPTION, subcommand.getActionVerb(), id), author,
+                    timestamp, guild,
+                    new ModAuditLogWriter.Attachment(NEW_CONTENT_FILE_NAME,
+                            Objects.requireNonNull(newContent)),
+                    new ModAuditLogWriter.Attachment(PREVIOUS_CONTENT_FILE_NAME,
+                            Objects.requireNonNull(previousContent)));
+
+            case EDIT_WITH_MESSAGE -> ModAuditLogWriter.write("Tag-Manage Edit with message",
+                    String.format(LOG_EMBED_DESCRIPTION, subcommand.getActionVerb(), id), author,
+                    timestamp, guild,
+                    new ModAuditLogWriter.Attachment(NEW_CONTENT_FILE_NAME,
+                            Objects.requireNonNull(newContent)),
```

Comment:
You already checked whenever it's null?

You do this for all attachments, and all it does is making your code unreadable, for fun?

Reply:
I believe SonarLint picks on that if I don't do it.

Reply:
Does IntelliJ pick on it?
If yes, I understand
If no, fuck sonarlint it just needs a brain.

I'd personally just ignore it since I know it's not true, you literally check or it's null above.
and even if I wasn't sure whenever it was null, Objects#requireNonNull throws an empty NPE in this case, I'd just let the method it runs throw the NPE with more info

Reply:
What I could do, is to make these parameters `@NotNull`, and to use a junk string like this `""` for cases where the parameter is not used. that would also get rid of the null checks. whats your opinion, @Tais993, @Zabuzard?

Reply:
I'm not sure, having an empty string sound like *odd* behaviour.

Reply:
thats indeed a bad idea. the null-detection tool is there to help u. if u bypass it on purpose, ur just fighting against urself, thats the worst thing u could do.

i doubt sonar picks up on this, its intellijs null-detection feature. so u can safely push it without the objects.requirenonnull. if the warning annoys u, lower its level to "suggestion".

Reply:
Both IntelliJ and Sonar doesn't like that, actually.

Reply:
Please show the sonar error message, cause I really doubt.

Reply:
duplicate with https://github.com/Together-Java/TJ-Bot/pull/296#discussion_r825322621